### PR TITLE
Add passive automation device handling

### PR DESCRIPTION
**Pull request by skl77**
Purpose: Integrate some low cost devices, like Intertechno; Elro etc.
- These devices have only a receive feature and no active (request/response) behaviour, so it was necessary to implement some features to configure the MIG devices completly passive from configuration file
- therefore, i need some more configuration parameters in MIG Interface
- at the end i added a driver class to implement some Elro; InterTechno devices over a connair IP gateway
planned: developing a full arduino 433/868 Mhz tranceiver to control some "low cost" automation devices and weather devices from oregon scientific

### CODE REVIEW

**Comment by genielabs** [comment]
Hi thanks for this, but you're using deprecated property "Modules" in your mig interface. Also the following code https://github.com/genielabs/HomeGenie/pull/107/files#diff-60b26291cea88c31d88e0b6adb232660L1701 should not be there as it is too specific for this particular implementation.
I know there are some files that still use cr/lf formatting, but this issue should not be carried out by your commit as these files are not touched by your interface implementation.

**Comment by skl77** [comment]
Hi Gene,
thanks for the feedback. Do you have an suggestion for me, how can i implement such an interface in permit way ?

**Comment by genielabs** [comment]
Hi @skl77 thanks for this contribution. Can you please make the pull request only contain files that are strictly related to your interface additions? It is currently pushing a 24 files changes.

File: HomeGenie/Service/HomeGenieService.cs [hunk 1]
```diff
@@ -1707,7 +1707,23 @@ private void LoadModules()
                 HomeGenie.Service.TsList<Module> modules = (HomeGenie.Service.TsList<Module>)serializer.Deserialize(reader);
                 //
                 foreach (var module in modules)
-                {
+                {                    
+                    if (module.Domain == "HomeAutomation.ConnAir")
```
Comment (genielabs) [inline HomeGenie/Service/HomeGenieService.cs]: @skl77 this is the old way HG used to create modules. Now this is not used anymore as modules are requested by using the MIGInterface method "GetModules".

File: MigFiles/MIG/Interfaces/Controllers/LircRemote.cs [hunk 2]
```diff
@@ -226,6 +226,11 @@ public string Domain
         }
 
         public List<MIGServiceConfiguration.Interface.Option> Options { get; set; }
+        
+        /// <summary>
+        /// get modules and module properties
+        /// </summary>
+        public List<InterfaceModule> Modules { get; set; }
```
Comment (genielabs) [inline MigFiles/MIG/Interfaces/Controllers/LircRemote.cs]: This is not used anymore, implement interface method *GetModules* instead.

File: MigFiles/MIG/Interfaces/EmbeddedSystems/RaspiGPIO.cs [hunk 3]
```diff
@@ -157,6 +157,11 @@ public string Domain
         }
 
         public List<MIGServiceConfiguration.Interface.Option> Options { get; set; }
+       
+        /// <summary>
+        /// get modules and module properties
+        /// </summary>
+        public List<InterfaceModule> Modules { get; set; }
```
Comment (genielabs) [inline MigFiles/MIG/Interfaces/EmbeddedSystems/RaspiGPIO.cs]: You're carrying old code in your repository. This also has been removed in current HG code.

File: MigFiles/MIG/Interfaces/EmbeddedSystems/Weeco4mGPIO.cs [hunk 4]
```diff
@@ -258,6 +258,11 @@ public string Domain
 
         public List<MIGServiceConfiguration.Interface.Option> Options { get; set; }
 
+        /// <summary>
+        /// get modules and module properties
+        /// </summary>
+        public List<InterfaceModule> Modules { get; set; }
+
```
Comment (genielabs) [inline MigFiles/MIG/Interfaces/EmbeddedSystems/Weeco4mGPIO.cs]: this piece of code too... it's coming from a very old revision.

File: MigFiles/MIG/Interfaces/HomeAutomation/ExampleDriver.cs [hunk 5]
```diff
@@ -102,6 +102,11 @@ public string Domain
 
         public List<MIGServiceConfiguration.Interface.Option> Options { get; set; }
 
+        /// <summary>
+        /// get modules and module properties
+        /// </summary>
+        public List<InterfaceModule> Modules { get; set; }
+
```
Comment (genielabs) [inline MigFiles/MIG/Interfaces/HomeAutomation/ExampleDriver.cs]: this too... coming from old revision, as for other similar following files.

**Comment by genielabs** [comment]
Hi @skl77 , MIG Interface Drivers can now be developed as stand-alone projects and can be bundled and shipped as a file that can be imported in HG for adding new interface drivers.
Read this message, since this apply to you also:
https://github.com/genielabs/HomeGenie/pull/142#issuecomment-93490333

**Pull request closed without merging**